umpost/conversion_driver_esm1p5.py

```diff
@@ -110,14 +122,70 @@ def find_matching_fields_files(dir_contents, fields_file_name_pattern):
     return fields_file_paths
 
 
-def convert_fields_file_list(fields_file_paths, nc_write_dir):
+def get_nc_filename(fields_file_name, date=None):
+    """
+    Format a netCDF output filename based on the input fields file name and
+    its date. Assumes fields_file_name follows ESM1.5's naming convention
+    '{5 char run_id}.pa{unit}{date encoding}`.
+
+    Parameters
+    ----------
+    fields_file_name: name of fields file to be converted.
+    date: tuple of form (year, month, day) associated with fields file data,
+          or None. If None, ".nc" will be concatenated to the original fields
+          file name.
+
+    Returns
+    -------
+    name: formated netCDF filename for writing output.
+    """
+    if date is None:
+        return f"{fields_file_name}.nc"
+
+    stem = fields_file_name[0:FF_UNIT_INDEX + 1]
+    unit = fields_file_name[FF_UNIT_INDEX]
```

Comment:
This suffices for this use case, but you may want to consider adding a `TODO` to switch over to regular expressions. They're more flexible (especially if this conversion ever needs to be done for other models/filename patterns), and I think they're also a bit clearer as to what you're doing (i.e., it would be more explicit what you're pulling out of the filename pattern and from where).

Reply:
Also forgot to mention that REs would give immediate feedback on whether or not `fields_file_name` conforms to the expected pattern.

Reply:
That's a good idea. I've added a `# TODO` comment and will make a new issue for this.